cmd/otel-allocator/Dockerfile

```diff
@@ -1,26 +1,34 @@
 # Build the target allocator binary
 FROM golang:1.19 as builder
 
-WORKDIR /app
-
-# Copy go mod and sum files
-COPY go.mod go.sum ./
-
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
-COPY . .
+# Copy the go source
+COPY main.go main.go
+COPY allocation/ allocation/
+COPY collector/ collector/
+COPY config/ config/
+COPY diff/ diff/
+COPY prehook/ prehook/
+COPY server/ server/
+COPY target/ target/
+COPY watcher/ watcher/
 
 # Build the Go app
 RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o main .
 
-######## Start a new stage from scratch #######
-FROM alpine:latest  
-
-RUN apk --no-cache add ca-certificates
-
-WORKDIR /root/
-
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
```

Comment:
Any chance this can be done `FROM scratch` instead?  Using `gcr.io/distroless` is likely going to cause security review headaches for me.  I think all we need is to copy certs and (maybe) tzdb from the bases `alpine` image.

Reply:
@Aneurysm9 I'll take a look and test it using KiND and OpenShift :D